File: Java/Language/Generics/TypeParamTest1/Program1.java
class Program {

	private static String select(int index, String first, String second) {
		if((index % 2) == 1)
			return first;
		return second;
	}

	private static double select(int index, double first, double second) {
		if((index % 2) == 1)
			return first;
		return second;
	}

	public static void main(String[] args) {
		if(args.length > 0){
			int s = Integer.parseInt(args[0]);
			String ss = select(s, "Monday", "Friday");
			System.out.printf("Selected String = %s%n", ss);
			double sd = select(s, 45.6, 76.5);
			System.out.printf("Selected double = %s%n", sd);
			//double sd2 = select(s, "9.5", 5.9);
		}
	}
}

File: dotnet/Language/Generics/TypeParamTest1/DemoApp/Program1.cs
class Program
{
    private static string Select(int index, string first, string second)
    {
        if((index % 2) == 1)
            return first;
        return second;
    }
    
    private static double Select(int index, double first, double second)
    {
        if((index % 2) == 1)
            return first;
        return second;
    }

    public static void Main(string[] args)
    {
        if(args.Length > 0)
        {
            int s = int.Parse(args[0]);
            string ss = Select(s, "Monday", "Tuesday");
            Console.WriteLine($"Selected string = {ss}");
            double sd = Select(s, 45.6, 34.5);
            Console.WriteLine($"Selected double = {sd}");
            //double ssd = Select(s, "Monday", 34.5);
        }
    }
}